ament_cmake_python/cmake/ament_python_install_package.cmake

```diff
@@ -116,51 +111,107 @@ setup(
   )
 
   if(ARG_SETUP_CFG)
-    add_custom_command(
-      OUTPUT "${build_dir}/setup.cfg"
-      COMMAND ${CMAKE_COMMAND} -E copy "${ARG_SETUP_CFG}" "${build_dir}/setup.cfg"
-      MAIN_DEPENDENCY "${ARG_SETUP_CFG}"
-    )
-    add_custom_target(${package_name}_setup ALL
-      DEPENDS "${build_dir}/setup.cfg"
+    add_custom_target(
+      ament_cmake_python_symlink_${package_name}_setup ALL
+      COMMAND ${CMAKE_COMMAND} -E create_symlink
+        "${ARG_SETUP_CFG}" "${build_dir}/setup.cfg"
     )
   endif()
 
-  if(NOT ARG_SKIP_COMPILE)
-    set(extra_install_args --compile)
+  add_custom_target(
+    ament_cmake_python_symlink_${package_name} ALL
+    COMMAND ${CMAKE_COMMAND} -E create_symlink
+      "${ARG_PACKAGE_DIR}" "${build_dir}/${package_name}"
+  )
+
+  set(install_dir "${CMAKE_INSTALL_PREFIX}/${PYTHON_INSTALL_DIR}")
+
+  if(NOT AMENT_CMAKE_SYMLINK_INSTALL)
+    # Install as flat Python egg to mimic https://github.com/colcon/colcon-core
+    # handling of pure Python packages for non-symlink installs.
+    if(NOT ARG_SKIP_COMPILE)
+      set(extra_install_args --compile)
+    else()
+      set(extra_install_args --no-compile)
+    endif()
+
+    # NOTE(hidmic): Allow setup.py install to build, as there is no way to
+    # determine the Python package's source dependencies for proper build
+    # invalidation.
+    install(CODE
+      "set(install_dir ${install_dir})
+       set(extra_install_args ${extra_install_args})
+       if(DEFINED ENV{DESTDIR} AND NOT \"\$ENV{DESTDIR}\" STREQUAL \"\")
+         list(APPEND extra_install_args --root \$ENV{DESTDIR})
+         file(TO_CMAKE_PATH \"\$ENV{DESTDIR}/\${install_dir}\" install_dir)
+       endif()
+       message(STATUS
+         \"Installing: ${package_name} as flat Python egg under \${install_dir}\")
+       file(TO_NATIVE_PATH \"${CMAKE_INSTALL_PREFIX}\" install_prefix)
+       file(TO_NATIVE_PATH \"${CMAKE_INSTALL_PREFIX}/bin\" install_scripts_dir)
+       execute_process(
+         COMMAND
+           \"${PYTHON_EXECUTABLE}\" setup.py install
+             --single-version-externally-managed
+             --install-scripts \${install_scripts_dir}
+             --prefix \${install_prefix}
+             --record install.log
+             \${extra_install_args}
+         WORKING_DIRECTORY \"${build_dir}\"
+         OUTPUT_QUIET
+       )"
+    )
   else()
-    set(extra_install_args --no-compile)
-  endif()
+    # NOTE(hidmic): Make sure build directory makes it into the PYTHONPATH.
+    # See https://github.com/colcon/colcon-core/pull/163 for further reference.
+    file(TO_NATIVE_PATH "${build_dir}" build_prefix)
+
+    if(NOT "${package_name}" STREQUAL "${PROJECT_NAME}")
+      set(hook_name "${package_name}_python_develop")
```

Comment:
is this ok?

I don't think line 182 will work in this case

Reply:
It is OK, what's not OK is line 182. Good catch!

Reply:
See f14c59b4dc67ad175467cb91a3efae24b044b7df.